pkg/kapis/cluster/v1alpha1/handler.go

```diff
@@ -311,6 +311,20 @@ func (h *handler) updateKubeConfig(request *restful.Request, response *restful.R
 		api.HandleBadRequest(response, request, fmt.Errorf("failed to validate member cluster configuration, err: %v", err))
 	}
 
+	// Check if the cluster is the same
+	kubeSystem, err := clientSet.CoreV1().Namespaces().Get(context.TODO(), metav1.NamespaceSystem, metav1.GetOptions{})
+	if err != nil {
+		api.HandleBadRequest(response, request, err)
+		return
+	}
+	if kubeSystem.UID != cluster.Status.UID {
```

Comment:
Be careful here, since `cluster.Status.UID` only added after v3.2.1, it may not work in previous versions.

Reply:
No need to worry since this `updateKubeConfig` function is also introduced after v3.2.1, I'm just perfecting it.